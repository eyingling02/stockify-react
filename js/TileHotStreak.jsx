const React = require('react')
const TileHotStreak = (props) => (
    <div className='outer'>
        <div className='buy'>
            <a href='javascript:;'>BUY</a>
        </div>
        <div className='inner'>
            <a href='livingroom.html' target='_blank'>
                <h2>{props.Ticker}</h2>
                <h3>{props.StreakInDays} predictions!</h3>
                <h3>{props.PredictionCount} days</h3>
                <h3>{props.Username}</h3>
            </a>
        </div>
        <div className='sell'>
            <a href='javascript:;'>SELL</a>
        </div>
    </div>
)

const {string} = React.PropTypes

// Tile.propTypes = {
//   title: string.isRequired,
//   description: string.isRequired,
//   year: string.isRequired,
//   poster: string
// }

module.exports = TileHotStreak
