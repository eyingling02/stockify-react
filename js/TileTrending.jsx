const React = require('react')
const TileTrending = (props) => (
    <div className='outer'>
        <div className='buy'>
            <a href='javascript:;'>UP</a>
        </div>
        <div className='inner'>
            <a href='livingroom.html' target='_blank'>
                <h2>{props.Symbol}</h2>
                <h2>{props.ChangePercent}%</h2>
                <h3>{props.CompanyName} days</h3>
            </a>
        </div>
        <div className='sell'>
            <a href='javascript:;'>DOWN</a>
        </div>
    </div>
)

const {string} = React.PropTypes

module.exports = TileTrending
