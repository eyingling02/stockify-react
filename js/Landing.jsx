import React from 'react'
import { Link } from 'react-router'
import TileHotStreak from './tile-hotstreak'
import TileTrending from './tile-trending'
import TileMyFavorites from './tile-myfavorites'
import TileFriendsOpinions from './tile-friendsopinions'
import TileCategory from './tile-category'

const data = require('../data')

const Landing = React.createClass({
    render() {
        return (
            <div id="picks">
                <div className="row">
                  <div className="category">
                    <p>My Favorites</p>
                  </div>
                  <div className="regular slider">
                    {data.MyFavorites.map((stock) => (<TileMyFavorites {...stock} key={stock.Symbol}/>))}
                  </div>
                </div>
                <div className="streaks row">
                    <div className="category">
                        <p>Hot Streaks</p>
                    </div>
                    <div className="regular slider">
                        {data.Hotstreak.map((stock) => (<TileHotStreak {...stock} key={stock.UserId}/>))}
                    </div>
               </div>
               <div className="row">
                 <div className="category">
                   <p>#Hashtags</p>
                 </div>
                 <div className="regular slider">
                   {data.FriendsOpinions.map((stock) => (<TileFriendsOpinions {...stock} key={stock.UserId}/>))}
                 </div>
               </div>
               <div className="row">
                 <div className="category">
                   <p>Shared News</p>
                 </div>
                 <div className="regular slider">
                   {data.FriendsOpinions.map((stock) => (<TileFriendsOpinions {...stock} key={stock.UserId}/>))}
                 </div>
               </div>
               <div className="row">
                 <div className="category">
                   <p>Friends Opinions</p>
                 </div>
                 <div className="regular slider">
                   {data.FriendsOpinions.map((stock) => (<TileFriendsOpinions {...stock} key={stock.UserId}/>))}
                 </div>
               </div>
                <div className="row">
                    <div className="category">
                        <p>Trending</p>
                    </div>
                    <div className="regular slider">
                      {data.TrendingHigher.map((stock) => (<TileTrending {...stock} key={stock.Symbol}/>))}
                    </div>
                </div>
                <div className="row">
                    <div className="category">
                        <p>Industry: Technology</p>
                    </div>
                    <div className="regular slider">
                      {data.Category.map((stock) => (<TileCategory {...stock} key={stock.Symbol}/>))}
                    </div>
                </div>
            </div>
        )
    }
})

export default Landing
