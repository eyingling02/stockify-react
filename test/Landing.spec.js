import React from 'react'
import Landing from '../js/Landing.jsx'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import TileTrending from '../js/TileTrending.jsx'
import TileHotStreak from '../js/TileHotStreak.jsx'
import TileMyFavorites from '../js/TileMyFavorites.jsx'
import TileFriendsOpinions from '../js/TileFriendsOpinions.jsx'
import preload from '../data.json'

test('Landing snapshot test', () => {
  const component = shallow(<Landing />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})

test('Landing page renders the correct number of stock tiles for the Hot Streaks category', () => {
  const component = shallow(<Landing />)
  expect(preload.Hotstreak.length).toEqual(component.find(TileHotStreak).length)
})

test('Landing page renders the correct number of stock tiles for the My Favorites category', () => {
  const component = shallow(<Landing />)
  expect(preload.MyFavorites.length).toEqual(component.find(TileMyFavorites).length)
})

test('Landing page renders the correct number of stock tiles for the Friends Opinions category', () => {
  const component = shallow(<Landing />)
  expect(preload.FriendsOpinions.length).toEqual(component.find(TileFriendsOpinions).length)
})

test('Landing page renders the correct number of stock tiles for the Trending Higher category', () => {
  const component = shallow(<Landing />)
  expect(preload.TrendingHigher.length).toEqual(component.find(TileTrending).length)
})
