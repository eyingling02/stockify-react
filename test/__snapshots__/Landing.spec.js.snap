exports[`test Landing snapshot test 1`] = `
<div
  id="picks">
  <div
    className="row">
    <div
      className="category">
      <p>
        My Favorites
      </p>
    </div>
    <div
      className="regular slider">
      <TileMyFavorites
        Duration={100}
        Symbol="UA"
        TodayGainLoss={1.29}
        TotalGainLoss={15.45}
        UserId={1} />
      <TileMyFavorites
        Duration={56}
        Symbol="FB"
        TodayGainLoss={4.5}
        TotalGainLoss={5.45}
        UserId={2} />
      <TileMyFavorites
        Duration={12}
        Symbol="NVDA"
        TodayGainLoss={1.21}
        TotalGainLoss={25.54}
        UserId={3} />
      <TileMyFavorites
        Duration={56}
        Symbol="MDT"
        TodayGainLoss={4.5}
        TotalGainLoss={5.45}
        UserId={2} />
      <TileMyFavorites
        Duration={12}
        Symbol="TWX"
        TodayGainLoss={1.21}
        TotalGainLoss={25.54}
        UserId={3} />
    </div>
  </div>
  <div
    className="streaks row">
    <div
      className="category">
      <p>
        Hot Streaks
      </p>
    </div>
    <div
      className="regular slider">
      <TileHotStreak
        PredictionCount={4}
        StreakInDays={8}
        Ticker="GILD"
        UserId={1}
        Username="thevictorchen" />
      <TileHotStreak
        PredictionCount={4}
        StreakInDays={8}
        Ticker="AMZN"
        UserId={2}
        Username="erica" />
      <TileHotStreak
        PredictionCount={4}
        StreakInDays={8}
        Ticker="GILD"
        UserId={3}
        Username="thenelson" />
      <TileHotStreak
        PredictionCount={4}
        StreakInDays={8}
        Ticker="AMZN"
        UserId={4}
        Username="thenelson" />
      <TileHotStreak
        PredictionCount={4}
        StreakInDays={8}
        Ticker="AMZN"
        UserId={5}
        Username="erica" />
    </div>
  </div>
  <div
    className="row">
    <div
      className="category">
      <p>
        Friends Opinions
      </p>
    </div>
    <div
      className="regular slider">
      <TileFriendsOpinions
        Opinion="Nothing good happening in their future. It\'s a toxic stock that lives with Zynga..."
        Symbol="UA"
        UserId={1}
        Username="nelsonchen" />
      <TileFriendsOpinions
        Opinion="Nothing good happening in their future. It\'s a toxic stock that lives with Zynga..."
        Symbol="FB"
        UserId={2}
        Username="nelsonchen" />
      <TileFriendsOpinions
        Opinion="Nothing good happening in their future. It\'s a toxic stock that lives with Zynga..."
        Symbol="NVDA"
        UserId={3}
        Username="nelsonchen" />
      <TileFriendsOpinions
        Opinion="Nothing good happening in their future. It\'s a toxic stock that lives with Zynga..."
        Symbol="MDT"
        UserId={4}
        Username="nelsonchen" />
      <TileFriendsOpinions
        Opinion="Nothing good happening in their future. It\'s a toxic stock that lives with Zynga..."
        Symbol="TWX"
        UserId={5}
        Username="nelsonchen" />
    </div>
  </div>
  <div
    className="row">
    <div
      className="category">
      <p>
        Trending
      </p>
    </div>
    <div
      className="regular slider">
      <TileTrending
        ChangePercent={8.94}
        CompanyName="Sprint Corporation"
        Symbol="S" />
      <TileTrending
        ChangePercent={6.86}
        CompanyName="Twitter"
        Symbol="TWTR" />
      <TileTrending
        ChangePercent={4.36}
        CompanyName="Valero Energy Corporation Commo"
        Symbol="VLO" />
      <TileTrending
        ChangePercent={4.28}
        CompanyName="United States Steel"
        Symbol="X" />
      <TileTrending
        ChangePercent={4.19}
        CompanyName="J.C. Penney Company"
        Symbol="JCP" />
      <TileTrending
        ChangePercent={4.05}
        CompanyName="CBS Corporation Class B Common"
        Symbol="CBS" />
      <TileTrending
        ChangePercent={3.98}
        CompanyName="Ford Motor Company"
        Symbol="F" />
      <TileTrending
        ChangePercent={3.96}
        CompanyName="American Eagle Outfitters"
        Symbol="AEO" />
      <TileTrending
        ChangePercent={3.93}
        CompanyName="Tesla Motors"
        Symbol="TSLA" />
      <TileTrending
        ChangePercent={3.69}
        CompanyName="Lions Gate Entertainment Corpor"
        Symbol="LGF" />
      <TileTrending
        ChangePercent={3.66}
        CompanyName="Groupon"
        Symbol="GRPN" />
      <TileTrending
        ChangePercent={3.51}
        CompanyName="General Motors Company Common S"
        Symbol="GM" />
      <TileTrending
        ChangePercent={3.38}
        CompanyName="Delta Air Lines"
        Symbol="DAL" />
      <TileTrending
        ChangePercent={3.37}
        CompanyName="Best Buy Co."
        Symbol="BBY" />
      <TileTrending
        ChangePercent={3.19}
        CompanyName="Mastercard Incorporated Common"
        Symbol="MA" />
    </div>
  </div>
  <div
    className="row">
    <div
      className="category">
      <p>
        Industry: Technology
      </p>
    </div>
    <div
      className="regular slider">
      <TileCategory
        ChangePercent={8.94}
        CompanyName="Sprint Corporation"
        Industry="Tech"
        Symbol="S" />
      <TileCategory
        ChangePercent={6.86}
        CompanyName="Twitter"
        Industry="Tech"
        Symbol="TWTR" />
      <TileCategory
        ChangePercent={4.36}
        CompanyName="Valero Energy Corporation Commo"
        Industry="Tech"
        Symbol="VLO" />
      <TileCategory
        ChangePercent={4.28}
        CompanyName="United States Steel"
        Industry="Tech"
        Symbol="X" />
      <TileCategory
        ChangePercent={4.19}
        CompanyName="J.C. Penney Company"
        Industry="Tech"
        Symbol="JCP" />
    </div>
  </div>
</div>
`;
